File: Claude 3 Haiku/C++/2-4.c
#include <iostream>
#include <vector>
using namespace std;

int binarySearch(const vector<int>& arr, int target) {
    int left = 0;
    int right = arr.size() - 1;

    while (left <= right) {
        int mid = left + (right - left) / 2;

        if (arr[mid] == target) {
            return mid;
        } else if (arr[mid] < target) {
            left = mid + 1;
        } else {
            right = mid - 1;
        }
    }

    return -1; // Target not found
}

int main() {
    vector<int> sortedList;
    int target, index;

    // Get the sorted list from the user
    cout << "Enter the sorted list (space-separated integers): ";
    int num;
    while (cin >> num) {
        sortedList.push_back(num);
    }
    cin.clear();
    cin.ignore(numeric_limits<streamsize>::max(), '\n');

    // Get the target value from the user
    cout << "Enter the target value: ";
    cin >> target;

    // Perform binary search and get the index
    index = binarySearch(sortedList, target);

    if (index == -1) {
        cout << "Target value not found in the list." << endl;
    } else {
        cout << "The target value is at index " << index << "." << endl;
    }

    return 0;
}

File: Claude 3 Haiku/python/2-4.py
def binary_search(sorted_list, target):
    left = 0
    right = len(sorted_list) - 1

    while left <= right:
        mid = (left + right) // 2

        if sorted_list[mid] == target:
            return mid
        elif sorted_list[mid] < target:
            left = mid + 1
        else:
            right = mid - 1

    return -1  # Target not found

def main():
    sorted_list = [int(x) for x in input("Enter the sorted list (space-separated integers): ").split()]
    target = int(input("Enter the target value: "))

    index = binary_search(sorted_list, target)

    if index == -1:
        print("Target value not found in the list.")
    else:
        print(f"The target value is at index {index}.")

if __name__ == "__main__":
    main()

